kedro_azureml/generator.py

```diff
@@ -157,7 +157,7 @@ def _get_versioned_azureml_dataset_name(
     def _get_input(self, dataset_name: str, pipeline: Pipeline) -> Input:
         if self._is_param_or_root_non_azureml_asset_dataset(dataset_name, pipeline):
             return Input(type="string")
-        elif dataset_name in self.catalog.list() and isinstance(
```

Comment:
catalog already has a `__contains__` dunder method that verify if a dataset_name fits either a dataset or a dataset_pattern (through regex).